lib/apiservers/engine/backends/system_portlayer.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vicbackends
+
+//****
+// system_portlayer.go
+//
+// Contains all code that touches the portlayer for system operations and all
+// code that converts swagger based returns to docker personality backend structs.
+// The goal is to make the backend code that implements the docker engine-api
+// interfaces be as simple as possible and contain no swagger or portlayer code.
+//
+// Rule for code to be in here:
+// 1. touches VIC portlayer
+// 2. converts swagger to docker engine-api structs
+// 3. errors MUST be docker engine-api compatible errors.  DO NOT return arbitrary errors!
+//		- Do NOT return portlayer errors
+//		- Do NOT return fmt.Errorf()
+//		- Do NOT return errors.New()
+//		- DO USE the aliased docker error package 'derr'
+
+import (
+	"fmt"
+	"net/http"
+
+	log "github.com/Sirupsen/logrus"
+	derr "github.com/docker/docker/errors"
+
+	"github.com/vmware/vic/lib/apiservers/portlayer/client/containers"
+	"github.com/vmware/vic/lib/apiservers/portlayer/client/misc"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+type ContainerCounts struct {
+	count      int
+	numRunning int
+	numStopped int
+	numPaused  int
+}
+
+// Interface to make unit testing viable
+type VicSystemProxy interface {
+	PingPortlayer() bool
+	ContainerCount() (ContainerCounts, error)
+	VCHInfo() (*models.VCHInfo, error)
+}
+
+type SystemProxy struct{}
```

Comment:
Only problem with that is that there the system backend has one systemproxy, and if we add counts to it, then dev needs to remember to call ContainerCount() before using the values.  By returning all the counts in a separate struct, the contract is this returns a point in time count of containers.